### ISSUE TITLE
Error reading a gazeteer is not visible to the user

### ISSUE THREAD

**Issue by dagosgi** [inception-project/inception#3545]
### Describe the bug

I am encountering some issues with recommenders that *should* be informed by a gazetteer. Basically, I have tried all that would come to my mind to make it work, but the internal recommender "string matcher" does not seem to recognize the .txt file I uploaded as a dictionary from which to retrieve terms and matching tags for annotation. Always active or not, with or without evaluation, trying to retrain the recommender every time by starting to insert some terms manually...still, it behaves as if the gazetteer wasn't there. The text file is formatted correctly, I double checked, and no error message showed up any time I deleted and reuploaded the gazetteer, nor when I tried deleting and recreating the feature the recommendation is based upon, or the whole layer.

### Expected behavior

Upon activation of a recommender system with an integrated gazetteer, I'd expect *some* recommendations to show up, based on the terms that can be found both in the dictionary and in the text. This doesn't happen.

### Screenshots

![image](https://user-images.githubusercontent.com/98110723/200792706-ad518ba5-693a-4a17-953f-bddcb304180b.png)

### Environment

Version and build ID: 25.1 (2022-10-11 21:15:38, build e85988ac)
Operating system: e.g. Windows
Browser: Chrome, Firefox

### Additional context

Here's a sample project with selected files and the dictionary we are supposed to use in the original project.
[esg-factors-gazetteer1408288947898743567.zip](https://github.com/inception-project/inception/files/9969271/esg-factors-gazetteer1408288947898743567.zip)

**Comment by reckart** [comment]
When trying to load the gazeteer, a message is logged that line 512 does not contain a tab character. After fixing that line, the project works.

**Comment by dagosgi** [comment]
Oh my...so the problem arised because that specific word at line 512 mistakenly didn't have any associated tag (?)
I went and fixed that, now that you told me...and apparently the problem is solved for real.

I have a follow-up question, then: now I'm feeling dumb, but _where_ could I have found out myself that **this** was the issue? I could not (and still cannot) find this information in the recommender log, nor (as I said before) any error message showed up upon upload of the dictionary. Where should I look?
Thank you
